rows.go

```diff
@@ -113,25 +111,31 @@ func (r *rows) Next(dest []driver.Value) error {
 			if err := col.Decode(&v); err != nil {
 				return err
 			}
-			dest[i] = v.Int64
+			if v.Valid {
+				dest[i] = v.Int64
+			}
 		case sppb.TypeCode_FLOAT64:
 			var v spanner.NullFloat64
 			if err := col.Decode(&v); err != nil {
 				return err
 			}
-			dest[i] = v.Float64
+			if v.Valid {
+				dest[i] = v.Float64
+			}
 		case sppb.TypeCode_NUMERIC:
 			var v spanner.NullNumeric
 			if err := col.Decode(&v); err != nil {
 				return err
 			}
-			dest[i] = v.Numeric
+			dest[i] = v
```

Comment:
Add `if v.Valid`?

Reply:
Yeah, I can see the confusion here. I've add a comment for why we are assigning `v` here in all cases (that is; also when `v.Valid` is false). The reason that we do so for `NUMERIC` and `DATE` data types is that the Go sql package does not have a native type mapping for these types. That means that if we were to return `nil` here, Go sql would not correctly understand what type it is. So instead we return a `NullNumeric` or `NullDate` in these cases.
The reason we don't do that for for example `INT64` is that the Go sql package natively maps the Spanner `INT64` data type to the Go `int64` type, and so we should return `nil` in those cases.